### PR TITLE
Baseline implementation for Side navigation

### LINKED ISSUE #100: Content Hub Header Iteration 4
This iteration will bring side navigation for content hub L2 pages
Screenshot 2024-02-07 at 18 57 38

### PR DESCRIPTION
**Pull request by saurabh-khare**
Side navigation introduced with desktop and mobile layout. It is present at hub level hence will be populated based on the content in nav.docx at the hub level. E.g. https://sap.sharepoint.com/:w:/r/sites/207899/_layouts/15/Doc.aspx?sourcedoc=%7B4EBF7745-8B58-438E-9377-7E54F8B617C2%7D&file=nav.docx
It can be overridden with 'sideNav' property in page metadata. If any of the reference is not present the side navigation will not be shown.

Limitations:
1. Fixed right margin for main content. This means side navigation will overlap with main content for width > 1600px. This will be addressed in #116 
2. Explore flyout menu not working in mobile layout. To be fixed after #116 

Fix #100 

Test URLs:
- Before: https://main--hlx-test--urfuwo.hlx.live/blog/2023/12/being-human-in-the-age-of-ai
- After: https://100-content-hub-header-iteration-4--hlx-test--urfuwo.hlx.live/blog/2023/12/being-human-in-the-age-of-ai

### CODE REVIEW

**Comment by saurabh-khare** [comment]
@benpeter If width >1600px then this is expected and will be fixed in #116. The side navigation is taking same gutters as top nav i.e. 246px (specified in figma) but the main content is not collapsing so side nav and main overlaps. If this appears for width < 1600px then its a bug and then please let me know this url.

**Comment by saurabh-khare** [comment]
`if (['topics', 'tags', 'author'].includes(area)) return;`
Instead of using url with hardcode white list is it possible to use **Content Type** metadata to figure out where to show side navigation?
Screenshot 2024-02-13 at 16 45 08
@pahupe @mhaack WDYT ?

**Comment by mhaack** [comment]
> `if (['topics', 'tags', 'author'].includes(area)) return;` Instead of using url with hardcode white list is it possible to use **Content Type** metadata to figure out where to show side navigation? Screenshot 2024-02-13 at 16 45 08 @pahupe @mhaack WDYT ?

@saurabh-khare where did you see that line?

**Comment by pahupe** [comment]
@saurabh-khare : that is totally fine with me, I just proposed a workaround to avoid having a sidenav painted onto certain pages types.
@mhaack : The line originates from a workaround (see my comment further up) I proposed to ensure we don't visually break existing pages. It's not in the code

**Comment by saurabh-khare** [comment]
> > `if (['topics', 'tags', 'author'].includes(area)) return;` Instead of using url with hardcode white list is it possible to use **Content Type** metadata to figure out where to show side navigation? Screenshot 2024-02-13 at 16 45 08 @pahupe @mhaack WDYT ?
> 
> @saurabh-khare where did you see that line?
It is not anywhere yet thats why we see side nav even on this page:
https://100-content-hub-header-iteration-4--hlx-test--urfuwo.hlx.page/tags/ai
Since /tags is not a physical url, header.js is able to resolve /tags/nav.plain.html and tries to show side nav. So what @pahupe  is suggesting is to create a logic of where not to show side nav. As of now now if any L1 directory can resolve nav.plain.html then side nav is shown. 
See header.js line 219

**Review by pahupe — COMMENTED** [review]
I have two things to add:
* the side nav is currently starting (vertically) at the same height as the hero banner. According to https://www.figma.com/file/oSetT4LbatRmXlcB2A7V8Z/Content-Hubs-2024?type=design&node-id=268-3637&mode=design&t=XbEERAkT6qiMhiOz-0, the side nav should start under the hero banner, and the hero banner should take full width
* The new grid is very cool, but it must be fine-tuned: Right now, the left margin + 3 columns (side nav) + 6 columns (body) are not correctly placed. The left and right margin should follow this spec: https://sap.frontify.com/document/223141#/layout/grid-breakpoints

**Review by pahupe — DISMISSED** [review]

**Comment by saurabh-khare** [comment]
@pahupe The margins for side nav are aligned with top nav as per figma spec (XS = 24px, S=48px, M&L=72px, XL=246px)
https://www.figma.com/file/OFEfvhyXJvYJ6MQZ30699m/v2.0-Brand-Digital-Design-System?type=design&node-id=8969%3A77845&mode=design&t=vADpjrj6zTiXNhRB-1

**Comment by saurabh-khare** [comment]
@sdmcraft Can you review as well and check if your suggestions from last review are fixed or not. Namely:
1. When clicking outside flyout menu the menu should collapse
2. The mobile menu should not overflow horizontally.

File: blocks/header/header.css [hunk 1]
```diff
@@ -359,7 +404,7 @@ header nav .nav-sections .nav-drop {
     padding: 12px 0;
     white-space: unset;
     text-wrap: wrap;
-    width: 30ch;
+    width: 22dvw;
```
Comment (sdmcraft) [inline blocks/header/header.css:407]: Looks like [Firefox still doesn't support `dvw`](https://caniuse.com/?search=dvw)

File: blocks/header/header.css [hunk 2]
```diff
@@ -463,29 +503,15 @@ header nav .nav-explore {
...
-
-header nav .nav-explore ul > li > a {
-  font-family: var(--udexTypographyNavigationRegularXLFontFamily);
-  font-weight: var(--udexTypographyNavigationRegularXLFontWeight);
-  color: var(--udexColorNeutralWhite);
-  font-size: var(--udexTypographyNavigationRegularXLFontSize);
-  line-height: var(--udexTypographyNavigationRegularXLLineHeight);
-  white-space: nowrap;
-  text-decoration: none;
+@media (width < 980px) {
```
Comment (sdmcraft) [inline blocks/header/header.css:508]: Usually `>=` queries should suffice and the smaller screen sizes are the defaults. Is it possible to achieve without the `<`

File: blocks/header/header.css [hunk 3]
```diff
@@ -463,29 +503,15 @@ header nav .nav-explore {
...
+  header nav .nav-explore ul > li > a {
+    color: var(--udexColorNeutralWhite);
+  }
 }
 
-@media (width > 980px) {
+@media (width >= 980px) {
```
Comment (sdmcraft) [inline blocks/header/header.css:514]: Nitpick: Maybe we can merge the media queries for the same view port. Usually I find them easier to manage this way.

File: blocks/header/header.css [hunk 4]
```diff
@@ -567,8 +600,173 @@ header nav .nav-actions button[aria-current='true']::after {
...
+  header aside .nav-side-home {
+    padding: var(--udexSpacer16) 0;
+    border-bottom: 1px solid var(--line-separator-color);
+    width: 90%;
+    box-shadow: unset;
+    border-radius: unset;
+  }
+}
+
+@media (width < 980px) {
```
Comment (sdmcraft) [inline blocks/header/header.css:657]: Same as prior comment on `<`

File: blocks/header/header.css [hunk 5]
```diff
@@ -567,8 +600,173 @@ header nav .nav-actions button[aria-current='true']::after {
...
+
+@media (width >=980px) {
+  header aside .nav-side-sections {
+    opacity: 1;
+    pointer-events: auto;
+    transform: unset;
+  }
+}
+
+@media (width >= 980px) {
```
Comment (sdmcraft) [inline blocks/header/header.css:701]: Same as prior comment on aggregating same view port media queries.

**Comment by sdmcraft** [comment]
> @sdmcraft Can you review as well and check if your suggestions from last review are fixed or not. Namely:
> 
> 1. When clicking outside flyout menu the menu should collapse
> 2. The mobile menu should not overflow horizontally

Sure @saurabh-khare . I can confirm that 1 is fixed. 2 is fixed as well if I resize the browser window but reproduces if I use thew chrom dev tools to emulate a smaller view port. Don't know the reason why it behaves differently but this is definitely a development only scenario since only happens with dev tools emulation. Attaching video for your reference

https://github.com/urfuwo/hlx-test/assets/1191451/11aa26ce-0589-471f-a967-42c65badfb05

**Review by pahupe — APPROVED** [review]
This is PR will (temporarily) introduce a sidenav to L3 pages. This unwanted effect will be countered in ticket #116, where the side nav will be moved to the L2 page template.

Let's move forward with this and focus on next ticket #116 to have clean L2 and L3 page template grids

File: blocks/header/header.css (hunk 1 above)
Comment (saurabh-khare) [reply blocks/header/header.css:407]: @sdmcraft It does :) What you're looking at is dvw() method not the unit. If you scroll down you'll see it is supported and now recommended over vw as it dynamically adjusts to viewport

**Comment by saurabh-khare** [comment]
> > @sdmcraft Can you review as well and check if your suggestions from last review are fixed or not. Namely:
> > 
> > 1. When clicking outside flyout menu the menu should collapse
> > 2. The mobile menu should not overflow horizontally
> 
> Sure @saurabh-khare . I can confirm that 1 is fixed. 2 is fixed as well if I resize the browser window but reproduces if I use thew chrom dev tools to emulate a smaller view port. Don't know the reason why it behaves differently but this is definitely a development only scenario since only happens with dev tools emulation. Attaching video for your reference
> 
>  Screen.Recording.2024-02-16.at.5.41.14.PM.mov

Yes somehow appears with dev tools. Though its a minor quirk I'll try to address it with next PR.

File: blocks/header/header.css (hunk 3 above)
Comment (saurabh-khare) [reply blocks/header/header.css:514]: I like to follow this approach for managing media queries in large CSS files :) 
https://stackoverflow.com/questions/11605761/how-should-css3-media-queries-be-managed

**Pull request merged**